### PR TITLE
Moved extension to new template

### PR DESCRIPTION
**Pull request by andrasguseo**

### CODE REVIEW

**Comment by cliffordp** [comment]
@andrasguseo 
Can you take a fresh look if this is ready for code review?
Also, I didn't see a version bump / date updated.

**Comment by andrasguseo** [comment]
@cliffordp 
I went through it and adjusted it to the recent template. Please check.

File: src/Settings_Helper.php [hunk 1]
```diff
@@ -131,7 +125,7 @@ public function filter_options( $fields, $tab ) {
 				$misc_heading = [
 					'tribeMiscSettings' => [
 						'type' => 'html',
-						'html' => '<h3>' . esc_html__( 'Miscellaneous Settings', 'tribe-ext-custom-all-events-url' ) . '</h3>',
+						'html' => '<h3>' . esc_html__( 'Miscellaneous Settings', PLUGIN_TEXT_DOMAIN ) . '</h3>',
```
Comment (juanfra) [inline src/Settings_Helper.php]: I don't see us defining this constant (`PLUGIN_TEXT_DOMAIN `)

File: src/Settings_Helper.php [hunk 2]
```diff
@@ -101,12 +96,11 @@ public function add_fields( $fields, $setting_tab, $neighboring_field = null, $a
...
-
 	/**
 	 * Remove a field from one of the tabs in WP Admin > Events > Settings
 	 *
 	 * @param string $field_key   Option key for your setting. Example: 'fancyOptionName'.
-	 * @param string $setting_tab Settings tab where this will be added. Example: 'display'.
+	 * @param string $setting_tab Settings tab from where this will be removed. Example: 'display'.
 	 */
 	public function remove_field( $field_key, $setting_tab ) {
 		$this->remove_fields[$setting_tab][] = $field_key;
```
Comment (juanfra) [inline src/Settings_Helper.php]:
```suggestion
		$this->remove_fields[ $setting_tab ][] = $field_key;
```
We normally add spaces around array keys when we use vars as keys. If we use the key directly (Like `$array['key']`) then it's not required to add spaces.

**Review by juanfra — COMMENTED** [review]
Looking good! Left a couple comments

**Comment by cliffordp** [comment]
@andrasguseo please ping me after addressing Juan's feedback, if you still want me to review then. Keep up the great work getting stuff done! 🎉

File: src/Settings_Helper.php (hunk 2 above)
Comment (andrasguseo) [reply src/Settings_Helper.php]: Oh cool, I didn't know. Thanks for teaching me something new. :)

**Comment by andrasguseo** [comment]
@cliffordp if you would be so kind... Thanks!

File: src/Settings.php [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+		 *
+		 * Recommended: the plugin text domain, with hyphens converted to underscores.
+		 * Is forced to end with a single underscore. All double-underscores are converted to single.
+		 *
+		 * @see get_options_prefix()
+		 *
+		 * @param string $options_prefix
+		 */
+		private function set_options_prefix( $options_prefix ) {
+			$options_prefix = $options_prefix . '_';
```
Comment (cliffordp) [inline src/Settings.php]: implement this recent fix from Extension Template: https://github.com/mt-support/tribe-ext-extension-template/commit/a8051889c8f2135c32ef9ba2bc0824dd0b7823dc

File: tribe-ext-custom-all-events-url.php [hunk 4]
```diff
@@ -44,6 +55,32 @@ public function construct() {
...
+		 * Get this plugin's options prefix.
+		 *
+		 * Settings_Helper will append a trailing underscore before each option.
+		 *
+		 * @see \Tribe\Extensions\CustomAllEventsUrl\Settings::set_options_prefix()
+		 *
+		 * @return string
+		 */
+		private function get_options_prefix() {
+			return (string) str_replace( '-', '_', 'tribe-ext-custom-all-events-url' );
```
Comment (cliffordp) [inline tribe-ext-custom-all-events-url.php]: Match Ext Template: https://github.com/mt-support/tribe-ext-extension-template/blob/master/src/Settings.php#L100

File: tribe-ext-custom-all-events-url.php [hunk 5]
```diff
@@ -122,5 +171,17 @@ function custom_all_events_url( $url ) {
...
+		 *
+		 * @return mixed
+		 */
+		public function get_custom_url() {
+
+			$settings = $this->get_settings();
+
+			return $settings->get_option( 'custom_all_events_url' );
+		}
+
```
Comment (cliffordp) [inline tribe-ext-custom-all-events-url.php]: delete all this

File: tribe-ext-custom-all-events-url.php [hunk 6]
```diff
@@ -113,7 +161,8 @@ public function add_settings() {
 		 * @return mixed
 		 */
 		function custom_all_events_url( $url ) {
-			$custom_url = tribe_get_option( $this->opts_prefix . 'custom_all_events_url' );
+
+			$custom_url = $this->get_custom_url();
```
Comment (cliffordp) [inline tribe-ext-custom-all-events-url.php]:
```suggestion

			$custom_url = $this->get_settings()->get_option( 'custom_all_events_url' );
```

**Review by cliffordp — CHANGES_REQUESTED** [review]
Fixes we discovered from earlier today on another extension

File: tribe-ext-custom-all-events-url.php (hunk 5 above)
Comment (andrasguseo) [reply tribe-ext-custom-all-events-url.php]: Deleted

File: tribe-ext-custom-all-events-url.php (hunk 6 above)
Comment (andrasguseo) [reply tribe-ext-custom-all-events-url.php]: Was outdated, couldn't commit, so changed it.

File: tribe-ext-custom-all-events-url.php [hunk 7]
```diff
@@ -44,6 +55,33 @@ public function construct() {
...
+		 * Get this plugin's options prefix.
+		 *
+		 * Settings_Helper will append a trailing underscore before each option.
+		 *
+		 * @see \Tribe\Extensions\CustomAllEventsUrl\Settings::set_options_prefix()
+		 *
+		 * @return string
+		 */
+		private function get_options_prefix() {
+			//return $this->settings->get_options_prefix;
```
Comment (cliffordp) [inline tribe-ext-custom-all-events-url.php]:
```suggestion
```

File: src/Settings.php [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+		 * Recommended: the plugin text domain, with hyphens converted to underscores.
+		 * Is forced to end with a single underscore. All double-underscores are converted to single.
+		 *
+		 * @see get_options_prefix()
+		 *
+		 * @param string $options_prefix
+		 */
+		private function set_options_prefix( $options_prefix = '' ) {
+			if ( empty( $opts_prefix ) ) {
+				$opts_prefix = str_replace( '-', '_', 'tribe-ext-extension-template' ); // The text domain.
```
Comment (cliffordp) [inline src/Settings.php]:
```suggestion
				$opts_prefix = str_replace( '-', '_', 'tribe-ext-custom-all-events-url' ); // The text domain.
```

File: src/Settings.php [hunk 9]
```diff
@@ -0,0 +1,150 @@
+<?php /** @noinspection PhpUndefinedVariableInspection */
```
Comment (cliffordp) [inline src/Settings.php]:
```suggestion
<?php
```

**Review by cliffordp — CHANGES_REQUESTED** [review]

File: src/Settings.php (hunk 8 above)
Comment (andrasguseo) [reply src/Settings.php]: 🤦

**Comment by andrasguseo** [comment]
This extension has been merged into The Events Calendar Extension: Tweaks
https://github.com/mt-support/tribe-ext-tec-tweaks

**Pull request closed without merging**